packages/core/src/ceramic.ts

```diff
@@ -465,17 +448,16 @@ class Ceramic implements CeramicApi {
   }
 
   /**
-   * Set DID provider
-   * @param provider - DID provider instance
+   * Sets the DID instance that will be used to author commits to documents. The DID instance
+   * also includes the DID Resolver that will be used to verify commits from others.
+   * @param did
    */
-  async setDIDProvider (provider: DIDProvider): Promise<void> {
-    this.context.provider = provider;
-    this.context.did = new DID({ provider, resolver: this.context.resolver })
-
-    if (!this.context.did.authenticated) {
-      await this.context.did.authenticate()
+  async setDID(did: DID): Promise<void> {
+    this.context.did = did
+    if (!did.authenticated) {
+      await this.did.authenticate()
```

Comment:
I think we should avoid doing this here. Instead we can do it inside of the *TileDocument* class on demand.